### DISCUSSION TITLE
Fonts in Linux

### DISCUSSION THREAD

**Discussion by the-solipsist** [Zettlr/Zettlr#3788] [General]
Here's what Zettlr looks like for me on Linux. Note the terrible font-kerning:

![image](https://user-images.githubusercontent.com/3080824/192140770-51e5b8a8-2bf0-43ac-a5ca-85fcc505a866.png)

![image](https://user-images.githubusercontent.com/3080824/192140855-abd26387-3ba2-45a6-a79a-e474eaa492af.png)

I'm wondering: is this true for everyone on all OSes? Is it true for all Linux users?  Is there some fix for this?

**Comment by nathanlesage** [comment]
Zettlr uses system fonts, the horrible kerning looks as if some font package is missing (I think there are some "default" font packages that can be installed, but I don't know the exact names...) Maybe it could also be due to an erroneous font ..?

What I can say, however, is that the fonts look perfectly fine on macOS and Windows.

**Reply by the-solipsist** [reply]
That's odd, since I'm using the standard OS font (Ubuntu).  I can confirm that changing the system font doesn't change the font in Zettlr. I now think it might have something to do with Electron. In some Electron apps (like Zettlr and WhatsDeck, which I just installed today), the font is a bit messed up. In other Electron apps, though (VSCodium and Atom), the fonts are fine. 

I think I'll search for Electron+Linux+Font issues and see if I come upon anything.

![image](https://user-images.githubusercontent.com/3080824/192166898-c9ce5263-d58c-4e53-8395-b8f6a272bfba.png)

**Reply by the-solipsist** [reply]
I think it would be useful to check if there are any flags relating to anti-aliasing or subpixel rendering that Zettlr could be enabling in Electron? It's clear that the problem is not with _all_ Electron apps, but some.

I tried adding `--disable-font-subpixel-positioning` to the exec line of the .desktop file (so `Exec=/opt/Zettlr/Zettlr --disable-font-subpixel-positioning  %U`), but that didn't help. I checked the .desktop files for VS Code / Chromium, and neither had that flag.

**Reply by nathanlesage** [reply]
CLI-flags don't work (except those we actually listen for), but what you could try changing is the CSS of the GUI itself, and see if you can find something that fixes the issue; I think there is a bunch of CSS rules out there so that you can try that out. I'm happy to implement anything that works to improve the Linux experience!

**Comment by the-solipsist** [comment]
Thanks. On the basis of the Zettlr docs, I tried things like
```
body #editor .CodeMirror {
    font-family: 'Noto Sans', 'monospace';
}
```
But that only changes the `body #editor`. What should I use to change the
font used in the side-bars, in the dialog boxes, in the window decorations,
etc.?

**Reply by nathanlesage** [reply]
There are multiple definitions of fonts across the app, unfortunately. I think the easiest way would be to search the source for `font-family:`, and then see which selector is each.
